File: MyTeletouch.Entities/CountryText.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace MyTeletouch.Entities
{
    public class CountryText : BaseModel
    {
        [Required]
        public int CountryId { get; set; }

        [Required]
        [MaxLength(2, ErrorMessage = "")]
        public string Locale { get; set; }

        [Required]
        [MaxLength(128, ErrorMessage = "")]
        public string Name { get; set; }

        [ForeignKey("CountryId")]
        public virtual Country Country { get; set; }
    }
}

File: MyTeletouch/Scripts/angular/models/CountryText.ts
/// <reference path="basemodel.ts" />

module Myteletouch {
    "use strict";

    export module Model {
        export class CountryText extends BaseModel {
            private _CountryId: number;
            private _Locale: string;
            private _Name: string;

            constructor() {
                super();

                this._CountryId = null;
                this._Locale = null;
                this._Name = null;
            }

            get Id(): number {
                return this.Id;
            }

            get CountryId(): number {
                return this._CountryId;
            }

            set CountryId(newValue: number) {
                this._CountryId = newValue;
            }

            get Locale(): string {
                return this._Locale;
            }

            set Locale(newValue: string) {
                this._Locale = newValue;
            }

            get Name(): string {
                return this._Name;
            }

            set Name(newValue: string) {
                this._Name = newValue;
            }
        }
    }
}